lrlex/src/lib/lexer.rs

```diff
@@ -337,8 +337,8 @@ impl<
                 i += longest;
             } else {
                 let current_state_id: Option<StartStateId> =
-                    StartStateId::try_from(current_state.id).ok();
-                lexemes.push(Err(LexError::new(
+                    Some(StartStateId::from(current_state.id));
```

Comment:
Fairly certain that we can change this to `StartStateId { _id: current_state.id }`, and remove the `From<usize>` impl,
now that this call-site is in the same crate as the structure.

Reply:
I did change this... and then realised that I marginally preferred `from` just because it abstracts away the `_` bit of `_id` (which we currently don't use). So I'm inclined to keep this as-is.

Reply:
I hope you don't mind if I make one last effort to dissuade this, because the `From` trait is `pub`, and `impl` has implicit visibility qualifiers, this makes it possible for users to construct a `StartStateId`, when really there is no benefit for having that public.

Maybe a compromise would be to move it into a non-pub `fn new` like below?

```
impl StartStateId {
    fn new(_id: usize) -> Self {
          Self { _id }
    }
}
```

Reply:
Very good point! Fixed in dcc5838.